scripts/nix-profile-daemon.sh.in

```diff
@@ -0,0 +1,54 @@
+# Only execute this file once per shell.
+if [ -n "$__ETC_PROFILE_NIX_SOURCED" ]; then return; fi
+__ETC_PROFILE_NIX_SOURCED=1
+
+# Set up secure multi-user builds: non-root users build through the
+# Nix daemon.
+if [ "$USER" != root -o ! -w @localstatedir@/nix/db ]; then
+    export NIX_REMOTE=daemon
+fi
+
+export NIX_USER_PROFILE_DIR="@localstatedir@/nix/profiles/per-user/$USER"
+export NIX_PROFILES="@localstatedir@/nix/profiles/default $HOME/.nix-profile"
+
+# Set up the per-user profile.
+mkdir -m 0755 -p $NIX_USER_PROFILE_DIR
+if test "$(stat -f '%u' $NIX_USER_PROFILE_DIR)" != "$(id -u)"; then
```

Comment:
Yep, if you have coreutils installed (even on Mac) it will give you:

```
stat: cannot read file system information for '%u': No such file or directory
WARNING: bad ownership on /nix/var/nix/profiles/per-user/...
stat: cannot read file system information for '%u': No such file or directory
WARNING: bad ownership on /nix/var/nix/gcroots/per-user/...
```

Maybe we should hardcode "/usr/bin/stat" to it, so that we get the MacOS/BSD version of stat?